pkg/netflow/portrollup/portrollup.go

```diff
@@ -33,8 +41,13 @@ const (
 // UseNewStoreAsCurrentStore is meant to be called externally to use new store as current store and empty the new store.
 type EndpointPairPortRollupStore struct {
 	portRollupThreshold int
-	curStore            map[string]*portRollupTracker
-	newStore            map[string]*portRollupTracker
+
+	// We might also use map[uint16]struct to store ports, but using []uint16 takes less mem.
+	// - Empty map is about 128 bytes
+	// - Empty list is about 24 bytes
+	// - It's more costly to search in a list, but the number of expected entry is at most equal to `portRollupThreshold`.
+	curStore map[string][]uint16
+	newStore map[string][]uint16
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏
This might be similar to @heyronhay's question, but I'm not sure I understand why we need to use `[]uint16` instead of a set (well a `map[uint16]struct{}` in go).
Is the memory impact really different?

Reply:
+1 for @FlorianVeaux's question.

Reply:
The map overhead is around 128 bytes based on https://stackoverflow.com/questions/15313105/memory-overhead-of-maps-in-go

See header: https://github.com/golang/go/blob/master/src/runtime/map.go#L115-L130

```
// A header for a Go map.
type hmap struct {
	// Note: the format of the hmap is also encoded in cmd/compile/internal/reflectdata/reflect.go.
	// Make sure this stays in sync with the compiler's definition.
	count     int // # live cells == size of map.  Must be first (used by len() builtin)
	flags     uint8
	B         uint8  // log_2 of # of buckets (can hold up to loadFactor * 2^B items)
	noverflow uint16 // approximate number of overflow buckets; see incrnoverflow for details
	hash0     uint32 // hash seed

	buckets    unsafe.Pointer // array of 2^B Buckets. may be nil if count==0.
	oldbuckets unsafe.Pointer // previous bucket array of half the size, non-nil only when growing
	nevacuate  uintptr        // progress counter for evacuation (buckets less than this have been evacuated)

	extra *mapextra // optional fields
}
```

And empty slice is about 24 bytes based on https://stackoverflow.com/questions/67839752/why-does-an-empty-slice-have-24-bytes

See header: https://github.com/golang/go/blob/master/src/runtime/slice.go#L15-L19

```
type slice struct {
	array unsafe.Pointer
	len   int
	cap   int
}
```

Since we are going to have multiple millions of entries, it can matter.

If we have 5M entry:
map: 128 * 5M => 640Mo
list: 24 * 5M => 120Mo

Also (and maybe more important), each entry in a map, is very likely going to cost more than each entry in a list.

Reply:
I think going from a struct with two maps to a single map will give you the memory savings needed, and the savings from map to list is probably not worth it to iterate through a list (memory is easier to get than CPU, moore's law is on life support for CPU but not for memory!).